tests/integration-tests/tests/common/schedulers_common.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from abc import ABCMeta, abstractmethod
+
+from retrying import retry
+
+from assertpy import assert_that
+from time_utils import minutes, seconds
+
+
+class SchedulerCommands(metaclass=ABCMeta):
+    """Define common scheduler commands."""
+
+    @abstractmethod
+    def __init__(self, remote_command_executor):
+        self._remote_command_executor = remote_command_executor
+
+    @abstractmethod
+    def assert_job_submitted(self, submit_output):
+        """
+        Assert that a job is successfully submitted.
+
+        :param submit_output: stdout from the submit command.
+        :return: the job id
+        """
+        pass
+
+    @abstractmethod
+    def wait_job_completed(self, job_id):
+        """
+        Wait for job completion.
+
+        :param job_id: id of the job to wait for.
+        :return: status of the job.
+        """
+        pass
+
+    @abstractmethod
+    def get_job_exit_status(self, job_id):
+        """
+        Retrieve the job exist status.
+
+        :param job_id: id of the job.
+        :return: the job exist status.
+        """
+        pass
+
+    @abstractmethod
+    def submit_command(self, command):
+        """
+        Submit a job to the scheduler.
+
+        :param command: command to submit.
+        :return: result from remote command execution.
+        """
+        pass
+
+    @abstractmethod
+    def assert_job_succeeded(self, job_id, children_number=0):
+        """
+        Assert that the job succeeded.
+
+        :param job_id: id of the job to check.
+        :param children_number: number of expected children. (e.g. array, multi-node)
+        """
+        pass
+
+
+class AWSBatchCommands(SchedulerCommands):
+    """Implement commands for awsbatch scheduler."""
+
+    def __init__(self, remote_command_executor):
+        super().__init__(remote_command_executor)
+
+    @retry(
+        retry_on_result=lambda result: "FAILED" not in result and any(status != "SUCCEEDED" for status in result),
+        wait_fixed=seconds(7),
+        stop_max_delay=minutes(15),
+    )
+    def wait_job_completed(self, job_id):  # noqa: D102
+        result = self._remote_command_executor.run_remote_command("awsbstat -d {0}".format(job_id))
```

Comment:
Do we need to run this remotely? Why not run locally and pass in the cluster name

Reply:
I decided to run these commands remotely for the following reasons:
* verify that the pcluster batch cli is correctly installed, configured, and upgraded on the master node
* verify that the default IAM role assumed by the master role has the required permissions to execute all batch commands.